### ISSUE TITLE
[java] Don't suggest replacing array with vargs if there are multiple array parameters

### ISSUE THREAD

**Issue by SUPERCILEX** [pmd/pmd#346]
**Rule Set:** design.html#UseVarargs

**Code Sample demonstrating the issue:**

```java
public void onRequestPermissionsResult(int requestCode, String[] permissions, int[] grantResults) { // Replacing just the "grantResults" array with vargs would be really confusing
```

**Running PMD through:** Gradle

**Comment by jsotuyod** [comment]
I would say only if the previous argument was an array. If the order was different I don't think it would be confusing and may be even beneficial.

Also, isn't your method an override for [this one](https://developer.android.com/reference/android/support/v4/app/ActivityCompat.OnRequestPermissionsResultCallback.html#onRequestPermissionsResult(int, java.lang.String[], int[]))? This shouldn't report on overrides (it should be reported on the interface, and only if it's part of the project and not a library).

Please, confirm you are actually overriding that method and if so:
1. are you adding the `@Override` annotation?
2. are you setting in any way the auxclasspath for type resolution? by default the android libs are not passed from Gradle to PMD, but plugins such as [this one](https://github.com/Monits/static-code-analysis-plugin) make sure they do.

There may be a separate issue here that may be worth reporting separately.

**Comment by SUPERCILEX** [comment]
@jsotuyod Well dang, I didn't know Android has an interface for that. No, I was just using a raw method with no overrides and I wasn't implementing anything. (Now that I know, I changed all my code and PMD is happy. Thanks! 😄). But still, I think PMD shouldn't complain if there are two arrays in a row, like you said.

**Comment by krichter722** [comment]
I agree with @SUPERCILEX  that this is a sometimes wanted behavior, but it should be a non-default option.
